README.md

```diff
@@ -1,33 +1,164 @@
-- [ ] ensure that Frontend team has the admin access to the repository
-- [ ] ensure that the Bots team has write access to the repository (this is needed to release the package)
-- [ ] provide usage examples in the `README.md`
-- [ ] enable the project on circleci.com to build and test your package
-- [ ] change the circleci status badge in this `README` to the new project
-- [ ] develop an awesome package
-- [ ] live long and prosper
-# example-pkg
-
-[![CircleCI](https://circleci.com/gh/smg-automotive/example-pkg/tree/main.svg?style=svg&circle-token=c183f151fea3c74453cf8dd962d31e115906a300)](https://circleci.com/gh/smg-automotive/example-pkg/tree/main)
+# api-client-pkg
+
+[![CircleCI](https://circleci.com/gh/smg-automotive/api-client-pkg/tree/main.svg?style=svg&circle-token=c183f151fea3c74453cf8dd962d31e115906a300)](https://circleci.com/gh/smg-automotive/example-pkg/tree/main)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 
 ## Usage
+
 ```
-npm install @smg-automotive/example
+npm install @smg-automotive/api-client-pkg
 ```
 
+````typescript
+import { ApiClient } from "@smg-automotive/api-client-pkg"
+````
+
+### Configuration
+
+The available instance methods are listed below. The specified config will be merged with the instance config.
+
+````typescript
+ApiClient.configure({
+  baseUrl: 'https://api.automotive.ch/api',
+  headers: { 'Content-Type': 'application/json' },
+})
+````
+
+### Instance methods
+
+The ApiClient provides promise-based methods for `GET`, `POST`, `PUT` and `DELETE`.
+
+### Typing
+
+You can define the request and the response type of your method as follows:
+
+````typescript
+interface Car {
+  make: string
+  horsepower: number
+}
+
+// data has type Car
+var data = await ApiClient.get<Car>('/listings/search')
+
+// data has type { id: string } and body is typed as Car
+var data = await ApiClient.post<{ id: string }, Car>('/listings/create',
+  { make: 'bmw', horsepower: 300 },)
+````
+
+### Parameters
+
+If your request URL contains parameters, you can use curly brackets and the ApiClient replaces it with the passed data.
+This is aligned with the OpenAPI specification used by Swagger. Hence, you can easily search for occurrences of your
+APIs.
+
+````typescript
+// fetch is called with https://baseUrl.api.ch/dealers/123/listings/456
+await ApiClient.get('dealers/{dealerId}/listings/{listingId}', {
+  params: { dealerId: 123, listingId: 456 },
+})
+````
+
+### Authorization
+
+You can pass `accessToken` to the request options if you are talking to a protected API. The client will automatically
+set the header with a Bearer token.
+
+````typescript
+// fetch is called with the header { Authorization: `Bearer ${accessToken}` }
+await ApiClient.post<{ id: number }, { name: string; listingIds: number[] }>(
```

Comment:
I dislike that we type the params in the projects, respectively for each api call. imagine we follow the silliest approach of just copy/pasting api calls around... we'll end up with a maintenance nightmare once the params types start diverging

with this approach to the api client, we really need ways to manage enforce good practices and discourage duplication in the projects

Reply:
If we have a generic API client it needs to support generics and dynamic type configuration. How we share and cluster the API calls in the projects is another story and shouldn't be a concern of the API client.

Reply:
I agree that if used wrong, it's risky. But I would argue that we need consistency in the project and not move the problem to the api-client. Otherwise we would have exactly the same issues as we had at CFY

Reply:
Yess, let's look into how we can enforce consistency in the projects on these